### PR TITLE
Core 7744 Add edit button to metadata view.

### PR DESCRIPTION
**Pull request by sri2k1us**
Since row editing is not supported with checkbox selection model, I have to implement a custom selection model for this.

### CODE REVIEW

File: ansible/playbooks/de-deploy-services.yaml [hunk 1]
```diff
@@ -376,7 +376,7 @@
       msg: ":heavy_check_mark: Done deploying infosquito"
 
 - name: Redeploy job-status-to-apps-adapter
-  hosts: jexevents
+  hosts: job-status-to-apps-adapter:&systems
```
Comment (psarando) [inline ansible/playbooks/de-deploy-services.yaml]: It looks like some of @ianmcorvidae's commits became part of this PR somehow...
Better rebase and force-push this branch to fix.

File: ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java [hunk 2]
```diff
@@ -329,6 +383,19 @@ public String getKey(Avu item) {
...
+            deleteMetadataButton.disable();
+            editMetadataButton.disable();
+        } else if (selectedSet.size() == 1) {
+            deleteMetadataButton.enable();
+            editMetadataButton.enable();
+        } else {
+            deleteMetadataButton.enable();
+            editMetadataButton.disable();
+        }
+    }
```
Comment (psarando) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java]: These buttons probably should be enabled or disabled depending on the `writable` boolean too.
So I would create 2 local booleans here, and call `deleteMetadataButton.setEnabled(deleteEnabled)` and `editMetadataButton.setEnabled(editEnabled)`, like we do elsewhere.

Comment (sri2k1us) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java]: 👍

File: ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java [hunk 3]
```diff
@@ -346,12 +407,17 @@ private String getUniqueAttrName(String attrName, int i) {
     }
 
     private void initUserMdGridEditor() {
-        userGridInlineEditing = new GridInlineEditing<>(userMdGrid);
-        userGridInlineEditing.setClicksToEdit(ClicksToEdit.TWO);
+        userGridRowEditing = new GridRowEditing<>(userMdGrid);
+        userGridRowEditing.setClicksToEdit(ClicksToEdit.TWO);
```
Comment (psarando) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java]: Will this still let a user double-click on a row to edit the AVU? What if the user doesn't have write permissions?

Comment (sri2k1us) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java]: `        if (writable) {
            initUserMdGridEditor();
            new QuickTip(userMdGrid);
        }`

Comment (sri2k1us) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java]: So it won't

**Comment by psarando** [comment]
I didn't catch any other issues :shipit:

**Comment by sri2k1us** [comment]
@psarando Thanks!

**Comment by sri2k1us** [comment]
@psarando thanks!

**Pull request merged**